project_1/include/packet.hpp

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <cstdint>
+
+struct TransmissionUnit {
+    explicit TransmissionUnit(std::uint64_t delivery_time_us);
+protected:
+    std::uint64_t estimated_delivery_time;
+};
+
+// Type for acknowledgement
+struct ACK : TransmissionUnit {
+    ACK(std::uint64_t receiver_id, std::uint64_t delivery_time_us);
+    std::uint64_t GetReceiverID() const;
+    std::uint64_t GetEstimatedDeliveryTime() const;
```

Comment:
Maybe it would be better for `TransmissionUnit` to have this method instead of implementing it twice here and in `PacketHeader`?